### PR TITLE
Restrict DX run height adjustment to only relevant glyph AND Correct PTY rendering on trailing half of fullwidth glyphs 

### LINKED ISSUE #2191: Chinese characters are not rendering correctly if I type above them.
# Environment

```none
Windows build number:  Microsoft Windows [Version 10.0.18362.239]
Windows Terminal version (if applicable):   0.2.1831.0
```

# Steps to reproduce

1. Fire up WSL
2. Run vim
3. Enter some Chinese characters, they seems to be fine.
4. Press Esc, press o to insert a new line above
5. Start typeing some English letter...
6. And the Chinese characters become smaller than normal size...
7. If I type English letter beneath, It works OK

# Expected behavior

Chinese characters should stay the same size no matter I type something above or beneath it

# Actual behavior

Chinese characters turn into smaller size  if I type English above the line.

### PR DESCRIPTION
**Pull request by miniksa**
## Summary of the Pull Request
- Height adjustment of a glyph is now restricted to itself in the DX
  renderer instead of applying to the entire run
- ConPTY compensates for drawing the right half of a fullwidth
  character. The entire render base has this behavior restored now as
  well.

## PR Checklist
* [x] Closes #2191
* [x] I work here
* [x] Tests added/passed
* [x] No doc
* [x] Am core contributor.

## Detailed Description of the Pull Request / Additional comments
Two issues:
1. On the DirectX renderer side, when confronted with shrinking a glyph,
   the correction code would apply the shrunken size to the entire run, not
   just the potentially individual glyph that needed to be reduced in size.
   Unfortunately while adjusting the horizontal X width can be done for
   each glyph in a run, the vertical Y height has to be adjusted for an
   entire run. So the solution here was to split the individual glyph
   needing shrinking out of the run into its own run so it can be shrunk.
2. On the ConPTY side, there was a long standing TODO that was never
   completed to deal with a request to draw only the right half of a
   two-column character. This meant that when encountering a request for
   the right half only, we would transmit the entire full character to be
   drawn, left and right halves, struck over the right half position. Now
   we correct the cursor back a position (if space) and draw it out so the
   right half is struck over where we believe the right half should be (and
   the left half is updated as well as a consequence, which should be OK.)

The reason this happens right now is because despite VIM only updating
two cells in the buffer, the differential drawing calculation in the
ConPTY is very simplistic and intersects only rectangles. This means
from the top left most character drawn down to the row/col cursor count
indicator in vim's modeline are redrawn with each character typed. This
catches the line below the edited line in the typing and refreshes it.
But incorrectly.

We need to address making ConPTY smarter about what it draws
incrementally as it's clearly way too chatty. But I plan to do that with
some of the structures I will be creating to solve #778.

## Validation Steps Performed
- Ran the scenario listed in #2191 in vim in the Terminal
- Added unit tests similar to examples given around glyph/text mapping
  in runs from Microsoft community page

### CODE REVIEW

File: src/renderer/vt/tracing.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void RenderTracing::TraceString(const std::string_view& instr) const
         const char* const seq = _seq.c_str();
         TraceLoggingWrite(g_hConsoleVtRendererTraceProvider,
                           "VtEngine_TraceString",
-                          TraceLoggingString(seq),
+                          TraceLoggingUtf8String(seq),
```
Comment (DHowett-MSFT) [inline src/renderer/vt/tracing.cpp]: ???

Comment (DHowett-MSFT) [reply src/renderer/vt/tracing.cpp]: you didn't actually do any other conpty things in this pull request, so the title confuses me

Comment (DHowett-MSFT) [reply src/renderer/vt/tracing.cpp]: oh it's the trimleft thing isn't it

Comment (miniksa) [reply src/renderer/vt/tracing.cpp]: yes it is.

Comment (miniksa) [reply src/renderer/vt/tracing.cpp]: also this was attempting to make it show up correct in tvpp. shhh, ignore it.

**Comment by DHowett-MSFT** [comment]
I've kicked the x64 build to restart only that leg.

**Comment by miniksa** [comment]
@msftbot merge in -1 minutes

**Pull request merged**